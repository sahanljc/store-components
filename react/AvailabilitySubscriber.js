export { default } from './components/AvailabilitySubscriber'
