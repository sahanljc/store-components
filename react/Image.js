import Image from './components/Image/index'

export default Image
