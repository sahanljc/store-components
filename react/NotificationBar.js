import NotificationBar from './components/NotificationBar/index'

export default NotificationBar
