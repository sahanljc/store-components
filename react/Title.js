import Title from './components/Title/index'

export default Title
