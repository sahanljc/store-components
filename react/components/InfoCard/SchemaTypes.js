export const textPostionValues = {
  LEFT: 'left',
  CENTER: 'center',
  RIGHT: 'right',
}

export const textAlignmentValues = {
  LEFT: 'left',
  CENTER: 'center',
  RIGHT: 'right',
}

export const callActionValues = {
  NONE: 'none',
  BUTTON: 'button',
  LINK: 'link',
}

export const textPositionTypes = {
  TEXT_POSITION_LEFT: {
    name: 'editor.info-card.textPosition.left',
    value: textPostionValues.LEFT,
  },
  TEXT_POSITION_CENTER: {
    name: 'editor.info-card.textPosition.center',
    value: textPostionValues.CENTER,
  },
  TEXT_POSITION_RIGHT: {
    name: 'editor.info-card.textPosition.right',
    value: textPostionValues.RIGHT,
  },
}

export const textAlignmentTypes = {
  TEXT_ALIGNMENT_LEFT: {
    name: 'editor.info-card.textAlignment.left',
    value: textAlignmentValues.LEFT,
  },
  TEXT_ALIGNMENT_CENTER: {
    name: 'editor.info-card.textAlignment.center',
    value: textAlignmentValues.CENTER,
  },
  TEXT_ALIGNMENT_RIGHT: {
    name: 'editor.info-card.textAlignment.right',
    value: textAlignmentValues.RIGHT,
  },
}

export const callToActionModeTypes = {
  CALL_ACTION_NONE: {
    name: 'editor.info-card.callAction.none',
    value: callActionValues.NONE,
  },
  CALL_ACTION_BUTTON: {
    name: 'editor.info-card.callAction.button',
    value: callActionValues.BUTTON,
  },
  CALL_ACTION_LINK: {
    name: 'editor.info-card.callAction.link',
    value: callActionValues.LINK,
  },
}
