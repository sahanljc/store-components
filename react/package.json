{
  "private": true,
  "scripts": {
    "pretest": "yarn",
    "test": "vtex-test-tools test",
    "lint": "eslint --ext js,jsx,ts,tsx ."
  },
  "dependencies": {
    "@material-ui/core": "^3.5.1",
    "@vtex/css-handles": "^1.0.2",
    "apollo-client": "^2.5.1",
    "classnames": "^2.2.5",
    "debounce": "^1.2.0",
    "downshift": "^1.31.14",
    "graphql": "~0.13.2",
    "graphql-tag": "~2.8.0",
    "prop-types": "^15.6.2",
    "ramda": "^0.25.0",
    "react": "^16.3.2",
    "react-apollo": "~2.5.2",
    "react-content-loader": "^3.1.2",
    "react-dom": "^16.8.3",
    "react-html-parser": "^2.0.2",
    "react-id-swiper": "^1.6.8",
    "react-intl": "^2.8.0",
    "react-resize-detector": "^3.0.1",
    "react-share": "^2.1.1",
    "react-slick": "^0.23.1",
    "sanitize-html": "^1.20.0",
    "slick-carousel": "^1.8.1"
  },
  "devDependencies": {
    "@vtex/test-tools": "^0.1.2",
    "apollo-cache-inmemory": "^1.4.3",
    "eslint": "^5.16.0",
    "eslint-config-vtex-react": "^4.0.0",
    "prettier": "^1.16.4",
    "typescript": "^3.4.1"
  },
  "vtexTestTools": {
    "defaultLocale": "en"
  }
}
