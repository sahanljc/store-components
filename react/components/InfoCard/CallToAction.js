import React from 'react'
import { Button } from 'vtex.styleguide'
import { Link } from 'vtex.render-runtime'
import styles from './infoCard.css'

import { callActionValues } from './SchemaTypes'

const ActionWrapper = ({ mode, text }) => {
  if (mode === callActionValues.BUTTON) {
    return <Button onClick={() => {}}>{text}</Button>
  }

  // Mode is link
  return <p className="link t-body b underline c-action-primary">{text}</p>
}

const CallToAction = ({ mode, text, url }) => {
  if (mode === callActionValues.NONE) {
    return null
  }

  return (
    <Link className={`${styles.infoCardCallActionContainer} mt6 mb6`} to={url}>
      <ActionWrapper text={text} mode={mode} />
    </Link>
  )
}

export default CallToAction
