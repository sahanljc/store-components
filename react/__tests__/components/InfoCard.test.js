import React from 'react'
import { render } from '@vtex/test-tools/react'

import InfoCard from './../../InfoCard'

describe('<InfoCard />', () => {
  const renderComponent = props => {
    const defaultProps = {
      isFullModeStyle: false,
      headline: 'MY HEADLINE',
      subhead: 'MY SUBHEAD',
      imageUrl: 'my-image.com/image.png',
      callToActionText: 'CLICK HERE',
      callToActionUrl: 'classic-shoes/p'
    }
    return render(<InfoCard {...defaultProps} {...props} />)
  }

  it('should render with full mode style', () => {
    const { asFragment, getByTestId } = renderComponent({
      isFullModeStyle: true,
    })
    expect(asFragment()).toBeDefined()
    const container = getByTestId('container')
    expect(container.style['background-image']).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render with image and text side by side', () => {
    const { asFragment, getByTestId } = renderComponent()
    expect(asFragment()).toBeDefined()
    const halfImage = getByTestId('half-image')
    expect(halfImage).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should render without a call to action', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
    })
    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should wrap half image in a link with imageActionUrl', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: false,
      callToActionMode: 'none',
      imageActionUrl: 'classic-shoes/p',
    })
    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
  it('should wrap whole container in a link with imageActionUrl', () => {
    const { asFragment } = renderComponent({
      isFullModeStyle: true,
      callToActionMode: 'none',
      imageActionUrl: 'classic-shoes/p',
      textAlignment: 'right',
      textPosition: 'center',
    })
    expect(asFragment()).toBeDefined()
    expect(asFragment()).toMatchSnapshot()
  })
})
