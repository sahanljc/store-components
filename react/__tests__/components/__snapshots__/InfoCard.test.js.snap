// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoCard /> should render with full mode style 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-start"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mh8-ns mh4-s w-40-ns items-start"
    >
      <h1
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base"
      >
        MY HEADLINE
      </h1>
      <p
        class="infoCardSubhead t-body mt6 c-on-base tl"
      >
        MY SUBHEAD
      </p>
      <a
        href="dummy"
      >
        <button>
          CLICK HERE
        </button>
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should render with image and text side by side 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center flex-ns flex-row bg-base ph2-ns pb2 justify-between"
    data-testid="container"
  >
    <div
      class="infoCardTextContainer flex flex-column w-50-ns ph3-s items-start pr4-ns"
    >
      <h1
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base"
      >
        MY HEADLINE
      </h1>
      <p
        class="infoCardSubhead t-body mt6 c-on-base tl"
      >
        MY SUBHEAD
      </p>
      <a
        href="dummy"
      >
        <button>
          CLICK HERE
        </button>
      </a>
    </div>
    <div
      class="w-50-ns"
    >
      <img
        data-testid="half-image"
        src="my-image.com/image.png"
        style="object-fit: cover;"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should render without a call to action 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-start"
    data-testid="container"
    style="background-image: url(my-image.com/image.png); background-size: cover;"
  >
    <div
      class="infoCardTextContainer flex flex-column mh8-ns mh4-s w-40-ns items-start"
    >
      <h1
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base"
      >
        MY HEADLINE
      </h1>
      <p
        class="infoCardSubhead t-body mt6 c-on-base tl"
      >
        MY SUBHEAD
      </p>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should wrap half image in a link with imageActionUrl 1`] = `
<DocumentFragment>
  <div
    class="infoCardContainer items-center flex-ns flex-row bg-base ph2-ns pb2 justify-between"
    data-testid="container"
  >
    <div
      class="infoCardTextContainer flex flex-column w-50-ns ph3-s items-start pr4-ns"
    >
      <h1
        class="infoCardHeadline t-heading-2 mt6 tl c-on-base"
      >
        MY HEADLINE
      </h1>
      <p
        class="infoCardSubhead t-body mt6 c-on-base tl"
      >
        MY SUBHEAD
      </p>
    </div>
    <div
      class="w-50-ns"
    >
      <a
        href="dummy"
      >
        <img
          data-testid="half-image"
          src="my-image.com/image.png"
          style="object-fit: cover;"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<InfoCard /> should wrap whole container in a link with imageActionUrl 1`] = `
<DocumentFragment>
  <a
    href="dummy"
  >
    <div
      class="infoCardContainer items-center bg-center bb b--muted-4 flex justify-center"
      data-testid="container"
      style="background-image: url(my-image.com/image.png); background-size: cover;"
    >
      <div
        class="infoCardTextContainer flex flex-column mh8-ns mh4-s w-40-ns items-center"
      >
        <h1
          class="infoCardHeadline t-heading-2 mt6 tc c-on-base"
        >
          MY HEADLINE
        </h1>
        <p
          class="infoCardSubhead t-body mt6 c-on-base tc"
        >
          MY SUBHEAD
        </p>
      </div>
    </div>
  </a>
</DocumentFragment>
`;
